lib/rpc/SingleJsonRpcProvider.ts

```diff
@@ -256,7 +256,20 @@ export class SingleJsonRpcProvider extends StaticJsonRpcProvider {
       startTimestampInMs: Date.now(),
     }
     try {
-      return await fn(...args)
+      // TODO: consolidate start time here and above
+      perf.startTimestampInMs = Date.now()
+      const result = await fn(...args)
+      perf.latencyInMs = Date.now() - perf.startTimestampInMs
```

Comment:
if the call fails this could be 0, right? since we are not longer updating the latencyInMs before logging?

Reply:
I think in that case, we won't log the latency metrics, because the code checks whether the perf object is successful or not https://github.com/Uniswap/routing-api/blob/main/lib/rpc/SingleJsonRpcProvider.ts#L163.

But admittedly, I think the implementation is a bit over-engineered, so I may choose to refactor/simplify later.